src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp

```diff
@@ -1,11 +1,20 @@
 #include "integration_loader.h"
 
 #include <exception>
+#include <unordered_set>
 
 #include "environment_variables.h"
 #include "logger.h"
 #include "util.h"
 
+template <>
+struct std::hash<trace::IntegrationMethod> {
+  std::size_t operator()(const trace::IntegrationMethod& integration_method) const noexcept {
+    // skip replacement operations while calculating hash
+    return std::hash<trace::WSTRING>()(integration_method.integration_name);
+  }
+};
```

Comment:
Not sure if I understand this correctly, but I think it would be less hacky and more readable to use `integrationMethods.find()` before `integrationMethods.push_back()` instead.

Reply:
I added comment, it should be less cryptic e521051b5955b0ae23cb47a175e6bcf5673f12aa

Reply:
I find C++ templates very complex. What would you have to do to get rid of them?

Reply:
Addressed during call. The original comment was confusing.